### PR TITLE
Nv32

### PR DESCRIPTION
**Pull request by manoj-freyr**
Changes to make specific confs for Navi32 and fix certain logging inconsistencies in IET reporting

### CODE REVIEW

**Comment by jkottiku** [comment]
Can you please provide a short description of the change in place of "Nv32" ?

File: src/rvs_blas.cpp [hunk 1]
```diff
@@ -437,8 +436,8 @@ bool rvs_blas::run_blass_gemm(std::string ops_type) {
                   int sol_index = 0;
                   int flags = 10;
 
-                  alpha = blas_alpha_val;
-                  beta  = blas_beta_val;
+                  alpha.data = blas_alpha_val;
+                  beta.data  = blas_beta_val;
```
Comment (TorreZuk) [inline src/rvs_blas.cpp:440]: This is wrong data conversion.  If you call hgemm you need to use a IEEE float16 representation.  You may need compiler flags and then just _Float16 alpha = (float)blas_alpha_value.     I add the cast as there was a bug in double to _Float16 in recent versions of the compiler.   Rocblas has a define ROCM_USE_FLOAT16 for internally not using the struct.  .data is an unsigned short so you would have serious data loss.

Comment (manoj-freyr) [reply src/rvs_blas.cpp:440]: will fix this

File: iet.so/src/iet_worker.cpp [hunk 2]
```diff
@@ -233,7 +233,7 @@ bool IETWorker::do_iet_power_stress(void) {
         rvs::lp::Log(msg, rvs::logtrace);
 
         //check whether we reached the target power
```
Comment (jkottiku) [inline iet.so/src/iet_worker.cpp:235]: Please update the comment as well.

**Review by jkottiku — CHANGES_REQUESTED** [review]

File: rvslib/CMakeLists.txt [hunk 3]
```diff
@@ -157,6 +157,8 @@ add_library(${RVS_TARGET}rt  ${SOURCES_RT})
 
 ## define rvslib library
 add_library(${RVS_TARGET}  ${SOURCES})
+target_compile_options(${RVS_TARGET} PRIVATE -mf16c)
+target_compile_definitions(${RVS_TARGET} PRIVATE ROCM_USE_FLOAT16)
 add_dependencies(${RVS_TARGET} ${RVS_TARGET}rt)
```
Comment (TorreZuk) [inline rvslib/CMakeLists.txt:162]: So  this define exposes:
#ifdef ROCM_USE_FLOAT16
typedef _Float16 rocblas_half;
#else
But using _Float16 is just as fine.

File: src/rvs_blas.cpp [hunk 4]
```diff
@@ -426,8 +425,8 @@ bool rvs_blas::run_blass_gemm(std::string ops_type) {
        }
 
       if(ops_type == "hgemm") {
-                  rocblas_float alpha;
-                  rocblas_float beta;
+                  //rocblas_half alpha;
+                  //rocblas_half beta;
```
Comment (TorreZuk) [inline src/rvs_blas.cpp:429]: These will be fine, but okay to do what you did below and just use _Float16.

**Review by TorreZuk — APPROVED** [review]
Very good it looks like you will have valid test data now (same requirements for both hgemm and gemm_ex for half).    Before you were getting only int truncation as the compiler converted the float to an unsigned int.

File: iet.so/src/iet_worker.cpp (hunk 2 above)
Comment (manoj-freyr) [reply iet.so/src/iet_worker.cpp:235]: done.

**Review by jkottiku — APPROVED** [review]

**Pull request merged**